homeassistant/components/tuya/binary_sensor.py

```diff
@@ -397,4 +397,8 @@ def is_on(self) -> bool:
         if isinstance(self.entity_description.on_value, set):
             return self.device.status[dpcode] in self.entity_description.on_value
 
-        return self.device.status[dpcode] == self.entity_description.on_value
+        state = self.device.status[dpcode]
+        if isinstance(state, str):
+            state = state.lower() in ["true", "on", "1"]
```

Comment:
This seems incorrect, if this is needed, it means the device is communicating an incorrect type for the datapoint. Do you have an example of this? And have you contact Tuya to report this issue for that specific device?

../Frenck

Reply:
As I shared in the previous screenshot, the API is returning a string value. If you see the screenshot, Tuya seems to be returning all values as a string. The device I am talking about is a smart plug 10A (category `cz`). In the screenshot below, you will see even parameters like `current`, `power`, `voltage` everything is returned as a string.

![image](https://github.com/user-attachments/assets/cdf6504d-9972-4b6b-acbb-6cc1e35e33d8)

This issue was raised by @amithkk in https://github.com/tuya/tuya-device-sharing-sdk/issues/15
It's been 3 months and Tuya is not responding.

Given the values being returned from the API as a string, being a truthy value, the integration reports everything with an "ON" value. I completely understand your point that it's not the right data type, but until Tuya responds and resolves the issue, we need to handle this edge case to ensure the integration works right (because this has been breaking automations as well every time HA restarts, or Tuya Integration is reloaded).

Reply:
> As I shared in the previous screenshot, the API is returning a string value.

Yes... understood that. Have you contacted Tuya.

> This issue was raised by @amithkk in https://github.com/tuya/tuya-device-sharing-sdk/issues/15

Cool, but have you contacted Tuya support about it?

Reply:
PS: If there is any place stuff like this could even be considered (although bad practice), it would be the upstream Python library.

Reply:
> > As I shared in the previous screenshot, the API is returning a string value.
> 
> Yes... understood that. Have you contacted Tuya.
> 
> > This issue was raised by @amithkk in [tuya/tuya-device-sharing-sdk#15](https://github.com/tuya/tuya-device-sharing-sdk/issues/15)
> 
> Cool, but have you contacted Tuya support about it?

No, personally I have not contacted Tuya Support for this, since there is already an issue related to this.

Reply:
> PS: If there is any place stuff like this could even be considered (although bad practice), it would be the upstream Python library.

I understand what you mean, and I agree. Is there a way to do this temporarily, since Tuya seems unresponsive? If at some point Tuya responds and fixes the issue, we can revert this change.